src/resource/file.ts

```diff
@@ -0,0 +1,157 @@
+//
+// Copyright 2022 Inrupt Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal in
+// the Software without restriction, including without limitation the rights to use,
+// copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the
+// Software, and to permit persons to whom the Software is furnished to do so,
+// subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+// INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
+// PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+// HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+// OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+// SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+//
+
+import {
+  UrlString,
+  getFile as coreGetFile,
+  overwriteFile as coreOverwriteFile,
+  saveFileInContainer as coreSaveFileInContainer,
+} from "@inrupt/solid-client";
+import { VerifiableCredential } from "@inrupt/solid-client-vc";
+import { fetchWithVc } from "../fetch";
+import { FetchOptions } from "../type/FetchOptions";
+
+/**
+ * Retrieve a File from a Solid Pod using an Access Grant to prove the caller
+ * is authorized to access the target resource.
+ *
+ * @param resourceUrl The URL of the target resource.
+ * @param accessGrant The Access Grant VC proving the caller is authorized.
+ * @param options Optional properties to customise the request behaviour.
+ * @returns A promise that resolves to a File if successful, and that rejects otherwise.
+ * @since 0.4.0
+ */
+export async function getFile(
+  resourceUrl: UrlString,
+  accessGrant: VerifiableCredential,
+  options?: FetchOptions
+): ReturnType<typeof coreGetFile> {
+  const fetchOptions: FetchOptions = {};
+  if (options && options.fetch) {
+    fetchOptions.fetch = options.fetch;
+  }
+
+  const authenticatedFetch = await fetchWithVc(
+    resourceUrl,
+    accessGrant,
+    fetchOptions
```

Comment:
Why not just 

```ts
const authenticatedFetch = await fetchWithVc(
    resourceUrl,
    accessGrant,
    { fetch: options?.fetch }
```

The current approach feels verbose and takes a few seconds to digest.

Reply:
This results in `fetch` being undefined:

<img width="562" alt="Screenshot 2022-08-31 at 16 19 29" src="https://user-images.githubusercontent.com/30827/187701467-c0acba01-668d-4c8a-9b33-95592707e8b9.png">

Which likely results in the method not working correctly. I would see this all as temporary though, until we release the major making fetch passing mandatory.